src/diric.jl

```diff
@@ -0,0 +1,78 @@
+export diric
+
+"""
+    kernel = diric(Ω::Real, n::Int)
+
+Dirichlet kernel, also known as periodic sinc function,
+where `n` should be a positive integer.
+Returns `sin(Ω * n/2) / (n * sin(Ω / 2))` typically,
+but `±1` when `Ω` is a multiple of 2π.
+
+In the usual case where 'n' is odd, the output is equivalent to
+``1/n \\sum_{k=-(n-1)/2}^{(n-1)/2} e^{i k Ω}``,
+which is the discrete-time Fourier transform (DTFT)
+of a `n`-point moving average filter.
+
+When `n` is odd or even, the function is 2π or 4π periodic, respectively.
+The formula for general `n` is
+`diric(Ω,n) = ``e^{-i (n-1) Ω/2}/n \\sum_{k=0}^{n-1} e^{i k Ω}``.
+
+When `Ω` is an `AbstractFloat` (e.g., `Float32` or `BigFloat`),
+the return type matches that of `Ω`.  Otherwise the return type is `Float64`.
+
+As of 2021-03-19, the Wikipedia definition has different factors.
+The definition here is consistent with scipy and other software frameworks.
+
+This implementation treats inputs near multiples of 2π fairly carefully.
+If the denominator `sin(Ω/2)` is close to 0, then this function uses
+an accurate 2nd-order Taylor expansion rather than simple division. 
+
+# Examples
+
+```jldoctest
+julia> round.(diric.((-2:0.5:2)*π, 5), digits=9)'
+1×9 adjoint(::Vector{Float64}) with eltype Float64:
+ 1.0  -0.2  0.2  -0.2  1.0  -0.2  0.2  -0.2  1.0
+
+julia> diric(0, 4)
+1.0
+```
+"""
+function diric(Ω::T, n::Int) where T <: AbstractFloat
+    n > 0 || throw(ArgumentError("n=$n is non-positive"))
+    sign = 1
+    if isodd(n)
+        Ω = rem2pi(Ω, RoundNearest) # [-π,π)
+    else
+        Ω = BigFloat(Ω ; precision = 2*precision(T))
+        Ω = 2 * rem2pi(abs(Ω)/2, RoundNearest) # [0,2π)
```

Comment:
```suggestion
        Ω = 2 * rem2pi(abs(Ω)/2, RoundNearest) # [-2π,2π)
```

Reply:
Note the `abs` above.  I did this so that I would not need the `else if` suggested below.
And I am pretty sure I tested accuracy near -2\pi to verify.

Reply:
But e.g.
```julia
julia> 2*rem2pi(4.0, RoundNearest)
-4.566370614359173
```
even though `4.0 > 0`.